Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Web3NameService.swift

```diff
@@ -9,23 +9,41 @@ import BraveShared
 import BraveCore
 
 extension BrowserViewController: Web3NameServiceScriptHandlerDelegate {
-  func web3NameServiceDecisionHandler(_ proceed: Bool, originalURL: URL, visitType: VisitType) {
+  func web3NameServiceDecisionHandler(_ proceed: Bool, web3Service: Web3Service, originalURL: URL, visitType: VisitType) {
     let isPrivateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
-    guard let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) else {
+    guard let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode),
+          let decentralizedDNSHelper = DecentralizedDNSHelper(
+            rpcService: rpcService,
+            ipfsApi: braveCore.ipfsAPI,
+            isPrivateMode: isPrivateMode
+          ) else {
       finishEditingAndSubmit(originalURL, visitType: visitType)
       return
     }
-    if proceed {
-      Task { @MainActor in
-        rpcService.setSnsResolveMethod(.enabled)
-        if let host = originalURL.host, let resolvedUrl = await resolveSNSHost(host, rpcService: rpcService) {
-          // resolved url
-          finishEditingAndSubmit(resolvedUrl, visitType: visitType)
+    Task { @MainActor in
+      switch web3Service {
+      case .solana:
```

Comment:
nit: These 2 cases can be merged ` case .solana, .ethereum`

Reply:
These are slightly different and can't be merged, `setSnsResolveMethod` vs `setEnsResolveMethod` 🙂